src/main/java/mate/academy/bookstorev2/service/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,88 @@
+package mate.academy.bookstorev2.service;
+
+import java.util.Objects;
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookstorev2.dto.cart.item.CartItemAddBookResponseDto;
+import mate.academy.bookstorev2.dto.cart.item.CartItemUpdateDto;
+import mate.academy.bookstorev2.dto.shopping.cart.AddToShoppingCartDto;
+import mate.academy.bookstorev2.dto.shopping.cart.ShoppingCartDto;
+import mate.academy.bookstorev2.exception.EntityNotFoundException;
+import mate.academy.bookstorev2.mapper.CartItemsMapper;
+import mate.academy.bookstorev2.mapper.ShoppingCartMapper;
+import mate.academy.bookstorev2.model.Book;
+import mate.academy.bookstorev2.model.CartItem;
+import mate.academy.bookstorev2.model.ShoppingCart;
+import mate.academy.bookstorev2.model.User;
+import mate.academy.bookstorev2.repository.book.BookRepository;
+import mate.academy.bookstorev2.repository.cart.item.CartItemRepository;
+import mate.academy.bookstorev2.repository.shopping.cart.ShoppingCartRepository;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final BookRepository bookRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemsMapper cartItemsMapper;
+
+    @Override
+    public ShoppingCart createShoppingCartForUser(User user) {
+        ShoppingCart cart = new ShoppingCart();
+        cart.setUser(user);
+        return shoppingCartRepository.save(cart);
+    }
+
+    @Override
+    public CartItemAddBookResponseDto addToCart(User user, AddToShoppingCartDto dto) {
+        ShoppingCart cart = shoppingCartRepository.findByUserId(user.getId())
+                .orElseGet(() -> createShoppingCartForUser(user));
```

Comment:
you create shop cart during the registration, if you can't find shop cart for user - throw exception